### PR TITLE
basic autentication task

### PR DESCRIPTION
**Pull request by Stelmaschyk**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/controller/AuthController.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Authentication", description = "Managing registration API")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/auth")
+public class AuthController {
+
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/controller/AuthController.java]:
```suggestion
```

File: src/main/java/mate/academy/bookstore/dto/book/UpdateBookRequestDto.java [hunk 2]
```diff
@@ -17,7 +16,7 @@ public class UpdateBookRequestDto {
     private String author;
     @Isbn
     private String isbn;
-    @NotNull
+    @NotBlank
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/dto/book/UpdateBookRequestDto.java]: NotBlank is used for String 
revert to NotNull for BigDecimal

File: src/main/java/mate/academy/bookstore/exception/CustomGlobalException.java [hunk 3]
```diff
@@ -47,4 +48,18 @@ private String getErrorMessage(ObjectError e) {
...
+            RegistrationException ex,
+            HttpHeaders headers,
+            HttpStatusCode status
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put(TIMESTAMP_FIELD, LocalDateTime.now());
+        body.put(STATUS_FIELD, HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getMessage().lines().toList();
+        body.put(ERRORS_FIELD, errors);
+        return new ResponseEntity<>(body, headers, status);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/exception/CustomGlobalException.java]:
```suggestion
        return new ResponseEntity<>(body, headers, HttpStatus.BAD_REQUEST);
```

File: src/main/java/mate/academy/bookstore/service/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/service/UserServiceImpl.java]:
```suggestion
        if (userRepository.existsByEmail(requestDto.getEmail())) {
```

File: src/main/java/mate/academy/bookstore/service/UserServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            throw new RegistrationException("This email is already registered");
+        }
+        User user = new User();
+        user.setEmail(requestDto.getEmail());
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/service/UserServiceImpl.java]: use MapStruct for this purpose

File: src/main/java/mate/academy/bookstore/service/UserServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        user.setPassword(requestDto.getPassword());
+        user.setFirstName(requestDto.getFirstName());
+        user.setLastName(requestDto.getLastName());
+        user.setShippingAddress(requestDto.getShippingAddress());
+        User savedUser = userRepository.save(user);
+        return userMapper.toUserDto(savedUser);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/service/UserServiceImpl.java]:
```suggestion
        userRepository.save(user);
        return userMapper.toUserDto(user);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/resources/db/changelog/changes/02-create-users-table.yaml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+                    nullable: false
+              - column:
+                  name: shipping_address
+                  type: varchar(255)
+              - column:
+                  name: is_deleted
+                  type: bit
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/changes/02-create-users-table.yaml]:
```suggestion
                    nullable: false
                    
```
Leave one empty line at the end of the file ;)

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 8]
```diff
@@ -1,3 +1,5 @@
 databaseChangeLog:
   - include:
       file: /db/changelog/changes/01-create-books-table.yaml
+  - include:
+      file: /db/changelog/changes/02-create-users-table.yaml
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: the same

**Review by liliia-ponomarenko — APPROVED** [review]
Awesome 😎

**Review by liliia-ponomarenko — APPROVED** [review]

**Review by ahoienko — APPROVED** [review]
Good job!

**Pull request merged**